### PR TITLE
Fix `StrongParameters#extract_value` to include blank values

### LINKED ISSUE #49695: url_helpers generate invalid urls if a records second composite primary key is empty
### Steps to reproduce

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", github: "rails/rails", branch: "main"
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :users, primary_key: ["firstname", "lastname"], force: true do |t|
    t.string "firstname", default: "", null: false
    t.string "lastname", default: "", null: false
  end
end

class User < ActiveRecord::Base
end

class TestApp < Rails::Application
  config.root = __dir__
  config.hosts << "example.org"
  config.secret_key_base = "secret_key_base"

  config.logger = Logger.new($stdout)
  Rails.logger = config.logger

  routes.draw do
    resources :users, only: [:show]
  end
end

require "action_controller/railtie"

class UsersController < ActionController::Base
  include Rails.application.routes.url_helpers

  def show
    @user = User.find(params.extract_value(:id))
    render json: @user
  end
end

class BugTest < Minitest::Test
  include Rack::Test::Methods
  include Rails.application.routes.url_helpers

  def test_with_both_keys
    user = User.create!(firstname: "John", lastname: "Doe")
    get user_path(user)
    assert last_response.ok?
  end

  def test_with_only_first_key
    user = User.create!(firstname: "Jack")
    get user_path(user)
    assert last_response.ok?
  end

  def test_with_only_second_key
    user = User.create!(lastname: "Black")
    get user_path(user)
    assert last_response.ok?
  end

  private

  def app
    Rails.application
  end
end
```

### Expected behavior

Generated urls should be handled correctly even if the composite primary key is partially empty. It might be design choice to not support empty keys but the url helpers should at least warn about invalid urls.

### Actual behavior

Generated urls which contain merged composite primary keys always respond with a 404 error if the respective record has a empty second primary key.

### System configuration
**Rails version**: 7.1.1

**Ruby version**: 3.2.2

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes #49695 (see there for the discussion).

### CODE REVIEW

**Review by nvasilevski — APPROVED** [review]
I think I know the answer but we are not opening any sort of ReDoS vulnerability by changing that right? Even if someone crafts an unusual url like `users/___________________________________________________` it will still be pretty quick to split and a really large `id` won't just fit into the URL

**Comment by fatkodima** [comment]
Yeah, good point! Just tested on 10k char string - instantaneous. It is not using regexp, so is O(n) of a string, which is fast and considering the url limit.

File: actionpack/lib/action_controller/metal/strong_parameters.rb [hunk 1]
```diff
@@ -963,12 +963,12 @@ def deep_dup
 
     # Returns parameter value for the given +key+ separated by +delimiter+.
     #
-    #   params = ActionController::Parameters.new(id: "1_123", tags: "ruby,rails")
+    #   params = ActionController::Parameters.new(id: "1_123", tags: "ruby,rails,")
```
Comment (byroot) [inline actionpack/lib/action_controller/metal/strong_parameters.rb:966]: I don't mind adding something that mention this edge case, but I don't think it should replace the existing example, as the first example in a doc should pretty much always be the happy path.

**Review by byroot — APPROVED** [review]
Just a quick comment on the documentation, but overall this makes sense to me.

File: actionpack/lib/action_controller/metal/strong_parameters.rb (hunk 1 above)
Comment (fatkodima) [reply actionpack/lib/action_controller/metal/strong_parameters.rb]: Agreed. Extracted as a separate note to the method's doc.

**Comment by byroot** [comment]
Backported as 407bed3f204fc8319d66d2eef8ce8ed162c445f4

**Pull request merged**